mapbox-data

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env node
+
+require('update-notifier')({ pkg: require('./package.json') }).notify();
+var MapboxDatasets = require('mapbox/lib/services/datasets');
+var getInput = require('./lib/get_input');
+var argv = require('yargs')
+    .usage('Usage: mapbox-data [options] <command> [parameters]')
+    .help('h')
+    .alias('h', 'help')
+    .alias('u', 'user')
+    .describe('u', 'Operate on the datasets for a particular user')
+    .command('create-dataset', 'Create a new dataset with a name and/or description')
+    .command('list-datasets', 'List the datasets belonging to a user')
+    .command('list-features', 'List the features in a dataset')
+    .command('get-dataset', 'Read information about a dataset')
+    .command('get-feature', 'Get information about a specific feature in a dataset')
+    .command('insert-features', 'Upload a geojson file to a dataset')
+    .command('replace-dataset', 'Delete all of the data in a dataset and replace with a geojson file')
+    .command('delete-feature', 'Delete a feature from a dataset')
+    .command('delete-dataset', 'Delete a dataset')
+    .demand(1, 3)
+    .example('mapbox-data create-dataset')
+    .example('mapbox-data insert-features id file.json')
+    .example('mapbox-data list-features id')
+    .argv;
+
+var opts = argv.user ? { account: argv.user } : {};
+var client = new MapboxDatasets(process.env.MapboxAccessToken, opts);
+
+var cmds = {
+    'insert-features': bulkInsert(argv._[1], argv._[2]),
+    'replace-dataset': bulkReplace(argv._[1], argv._[2]),
+    'create-dataset': client.createDataset.bind(client, {name:argv._[1], description:argv._[2]}),
+    'list-features': client.listFeatures.bind(client, argv._[1]),
+    'list-datasets': client.listDatasets.bind(client),
+    'delete-feature': client.deleteFeature.bind(client, argv._[2], argv._[1]),
+    'delete-dataset': client.deleteDataset.bind(client, argv._[1]),
+    'get-feature': client.readFeature.bind(client, argv._[2], argv._[1]),
+    'get-dataset': client.readDataset.bind(client, argv._[1]),
+};
+
+cmds[argv._[0]](handler);
+
+function handler(err, result) {
+    if (err || result === undefined) return console.error(err);
```

Comment:
In a CLI environment, its nice to make sure that the process exits with a non-zero exit code in the event of a failure. You can do that one of two ways;
```js
// if you want stack traces
if (err) throw err;
// ... or if you don't ...
if (err) {
  console.error(err.message || 'An unexpected error occurred');
  process.exit(1);
}
```

I would also suggest you differentiate these conditions:

```js
if (err) { ... return the information in the error object ... }
else if (!result) { ... inform the user that something unexpected happened ... }
```